topsort-api-v2.yml

```diff
@@ -466,6 +515,230 @@ components:
+          example: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
+        placement:
+          $ref: '#/components/schemas/Placement'
+        occurredAt:
+          type: string
+          format: date-time
+          description: RFC3339 formatted timestamp including UTC offset.
+          example: '2009-01-01T12:59:59-05:00'
+        opaqueUserId:
+          type: string
+          description: >
+            The opaque user ID allows correlating user activity, whether or not they are actually logged in.
+            It must be long lived.
+          example: 71303ce0-de89-496d-8270-6434589615e8
+        id:
+          type: string
+          description: >
+            The marketplace's unique ID for the impression.
+            This field ensures the event reporting is idempotent in case there is a network issue and the request is retried.
+            If there is no impression model on the marketplace side, generate a unique string that does not change if the event is resent.
+          example: eb874c98-bf4d-40a9-ae6d-fcf4cecb535c
+    Click:
+      description: >
+        A click is sent to Topsort when the consumer has clicked on an ad.
+        Topsort charges the vendor and pays the marketplace for clicks on ads in promoted placements on the marketplace app.
+      type: object
+      required:
+        - placement
+        - occurredAt
+        - opaqueUserId
+        - id
+      properties:
+        resolvedBidId:
+          type: string
+          description: If the click is over an ad promotion, this is the `resolvedBidId` field received from the /auctions request.
+          example: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
+        placement:
+          $ref: '#/components/schemas/Placement'
+        occurredAt:
+          type: string
+          format: date-time
+          description: RFC3339 formatted timestamp including UTC offset.
+          example: '2009-01-01T12:59:59-05:00'
+        opaqueUserId:
+          type: string
+          description: >
+            The opaque user ID allows correlating user activity, whether or not they are actually logged in.
+            It must be long lived.
+          example: 71303ce0-de89-496d-8270-6434589615e8
+        id:
+          type: string
+          description: >
+            The marketplace's unique ID for the click.
+            This field ensures the event reporting is idempotent in case there is a network issue and the request is retried.
+            If there is no click model on the marketplace side, generate a unique string that does not change if the event is resent.
+          example: b39d39ed-ea0e-4059-9d15-4990b39c85a2
+    Purchase:
+      description: >
+        A purchase is sent to Topsort once the user makes a payment transaction and it has been confirmed by the payment provider or financial entity in charge processing the transaction.
```

Comment:
Not sure if we want to add that restriction about having to validate the payment, as that may restrict where the event can be reported.

For example we may want to report the purchase after clicking.

Also, reporting the purchase when the payment has been validated affects the metrics, as there will be considerable delay.